server/bin/basket-proxy-server.js

```
@@ -18,40 +18,54 @@ var logger = require('mozlog')('server.basketproxy');
 
 var express = require('express');
 var bodyParser = require('body-parser');
+var cors = require('cors');
 
+var CORS_ORIGIN = config.get('public_url');
 var API_KEY = config.get('basket.api_key');
 var API_URL = config.get('basket.api_url');
 var VERIFY_URL = config.get('oauth_url') + '/v1/verify';
 
 // Verify an OAuth token and return the associated credentials
-function verifyOAuthToken(token) {
-  var defer = p.defer();
+function verifyOAuthToken(cb) {
+  return function (req, res, next) {
+    var authHeader = req.headers && req.headers.authorization;
 
-  request.post({
-    url: VERIFY_URL,
-    json: {
-      token: token
+    if (! authHeader) {
+      logger.error('unauthorized');
+      return next(new Error('Not authorized'));
```

This should raise as an "Unexpected error" for the user.